vocabularies/Org.OData.Authorization.V1.xml

```diff
@@ -123,6 +123,7 @@
       </ComplexType>
 
       <ComplexType Name="OAuth2Implicit" BaseType="Authorization.OAuthAuthorization">
+        <Annotation Term="Core.Description" String="Security note: OAuth2 implicit grant is considered to be not secure and should not be used by clients, see [OAuth 2.0 Security Best Current Practice](https://datatracker.ietf.org/doc/html/draft-ietf-oauth-security-topics.html#name-implicit-grant)." />

```

Comment:
```suggestion
        <Annotation Term="Core.Description" String="Security note: OAuth2 implicit grant should only be used by clients in combination with [PKCE](https://www.rfc-editor.org/rfc/rfc7636), see [OAuth 2.0 Security Best Current Practice](https://datatracker.ietf.org/doc/html/draft-ietf-oauth-security-topics.html#name-implicit-grant)." />
```
I think that implicit grant with PKCE is acceptable.

Reply:
https://owasp.org/www-project-web-security-testing-guide/latest/4-Web_Application_Security_Testing/05-Authorization_Testing/05-Testing_for_OAuth_Weaknesses states that

> Two flows will be deprecated in the release of [OAuth2.1](https://oauth.net/2.1/), and their usage is not recommended:
> * [Implicit Flow*](https://oauth.net/2/grant-types/implicit/): PKCE’s secure implementation renders this flow obsolete. Prior to PKCE, the implicit flow was used by client-side applications such as [single page applications](https://en.wikipedia.org/wiki/Single-page_application) since [CORS](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS) relaxed the [same-origin policy](https://developer.mozilla.org/en-US/docs/Web/Security/Same-origin_policy) for sites to inter-communicate. For more information on why the implicit grant is not recommended, review this [section](https://datatracker.ietf.org/doc/html/draft-ietf-oauth-security-topics#section-2.1.2).

Reply:
Is the PKCE flow described in https://www.rfc-editor.org/rfc/rfc7636 recommended? Is that the recommended successor of the implicit flow?

Reply:
Good questions, let's ask a security expert.

Reply:
The experts recommend PKCE, does this mean that we encourage [browser-based OAuth 2.0 clients](https://datatracker.ietf.org/doc/html/draft-ietf-oauth-browser-based-apps#name-browser-based-oauth-20-clie)?